portfolio/src/main/webapp/js/comments.js

```diff
@@ -108,3 +123,32 @@ function postComment(event) {
           "alert-danger");
     });
 }
+
+/**
+ * Gets a function that sends a comment delete request.
+ * @param {number} id
+ * @return Returns a function that deletes the comment associated with an ID.
+ */
+function getCommentDeleter(id) {
+  return () => {
+    const deleteUrl = 
+        new URL("/comments", `${location.protocol}//${location.hostname}`);
+    deleteUrl.searchParams.set("id", id);
+
+    fetch(deleteUrl, { method: 'DELETE' }).then((resp) => {
+        if (resp.ok) {
+          addNotification("Comment deleted successfully.", "alert-success");
+          loadComments();
+        } else {
+          return resp.text().then(
+              text => Promise.reject(`Error ${resp.status}: ${text}`));
+        }
+      }).catch(err => {
+        console.log(err);
+
+        addNotification(
+          "Failed to delete the comment! Please try again later.", 
+          "alert-danger");
+      });
```

Comment:
If I understand this code correctly the catch statement will only be called if and only if the response is not ok. In that case you should put the error handling logic in the else statement. This reduces the ammount of code significantly while accomplishing the same end.

Keep in mind that the suggestion below may not be correct javascript. It is merely to show you what I'm talking about.
```suggestion
        if (resp.ok) {
          addNotification("Comment deleted successfully.", "alert-success");
          loadComments();
        } else {
          console.log(`Error ${resp.status}: ${resp.text()}`);
          addNotification(
           "Failed to delete the comment! Please try again later.", 
           "alert-danger");
        }
      })
```

Reply:
The catch is also called in the case that there is an error from the request itself, like a failure to connect, invalid URL, etc.

However, I still agree with what you are saying and in the event that I encounter something like what you describe I'll avoid the catch block.